### ISSUE TITLE
JPG resolution drops to 96 dpi

### ISSUE THREAD

**Issue by AdiLew** [lovell/sharp#3017]
Hi,
I have a few scripts that I'm using to flip and rotate JPG files.
Below is the rotating function:
````
async function rotateImage(targetFilePath, deg = 0) {
    const sharpImage = sharp(targetFilePath)
    sharpImage.rotate(deg);
    const buffer = await sharpImage.toBuffer();
    await sharp(buffer).toFile(targetFilePath)
}
````
The image I'm processing is available here:
[https://drive.google.com/file/d/1WmqFj14LaUA0OAgWiiTgWUiTfLDaCqwj/view?usp=sharing](url)

The rotation works well, but it seems that the images lose resolution and drop from 600 dpi to 96 dpi.
Is there a way I can keep the original image's resolution while processing it?

Thanks!

**Comment by lovell** [comment]
JPEG doesn't have a concept of resolution, but this image has EXIF metadata:
```
$ vipsheader -a I600.jpg | grep Resolution
exif-ifd0-XResolution: 600000/1000 (600.000, Rational, 1 components, 8 bytes)
exif-ifd0-YResolution: 600000/1000 (600.000, Rational, 1 components, 8 bytes)
exif-ifd0-ResolutionUnit: 2 (Inch, Short, 1 components, 2 bytes)
```
Please see https://sharp.pixelplumbing.com/api-output#withmetadata for control over this.

**Comment by AdiLew** [comment]
I've tried adding the withMetadata() method, to keep all original exif like so, but it doesn't seem to work.
````
async function rotateImage(targetFilePath, deg = 0) {
    const sharpImage = sharp(targetFilePath).withMetadata();

    sharpImage.rotate(deg);
    const buffer = await sharpImage.toBuffer();
    await sharp(buffer).toFile(targetFilePath)
}
````

I've also tried dictating a hardcoded value for the X and Y resolution, as attached below, but the result is still  96dpi:
````
async function rotateImage(targetFilePath, deg = 0) {
    const exifObject = {
        exif: {
            IFD0: {
                image: {
                    XResolution: 600,
                    YResolution: 600
                }
            }
        }
    };
    const sharpImage = sharp(targetFilePath).withMetadata(exifObject);
    console.dir(sharpImage);
    sharpImage.rotate(deg);
    const buffer = await sharpImage.toBuffer();
    await sharp(buffer).toFile(targetFilePath)
}
````

Is there a working example for retaining both of these values?

**Comment by lovell** [comment]
Did you see the example at https://sharp.pixelplumbing.com/api-output#withmetadata 

```js
// Set output metadata to 96 DPI
const data = await sharp(input)
  .withMetadata({ density: 96 })
  .toBuffer();
```
(I need to fix the formatting of this example in the docs as it's currently being displayed as part of the previous example.)

**Comment by AdiLew** [comment]
I did, neither the density property did not seem to have any affect over the output file's x and y resolution.
I also looked in the [output unit tests](https://github.com/lovell/sharp/blob/master/lib/output.js) and tries to imitate the way the exif should have been dictated, but it didn't seem to work either.
Is this the proper way to describe the exif data?
````
{
        exif: {
            IFD0: {

                XResolution: '600dpi',
                YResolution: '600dpi'
            }
        }
````
